tasks/hyrax-metadata-updates/index.js

```diff
@@ -0,0 +1,290 @@
+const cloneDeep = require('lodash.clonedeep');
+
+const {
+  CMR
+} = require('@cumulus/cmr-client');
+
+const { isECHO10File, isUMMGFile, isCMRFilename } = require('@cumulus/cmrjs/cmr-utils');
+
+
+const { validateUMMG } = require('@cumulus/cmr-client/UmmUtils');
+const validate = require('@cumulus/cmr-client/validate');
+const ValidationError = require('@cumulus/cmr-client/ValidationError');
+
+const {
+  getS3Object,
+  s3PutObject,
+  parseS3Uri
+} = require('@cumulus/aws-client/S3');
+
+const libxmljs = require('libxmljs');
+
+/**
+ * generateAddress
+ *
+ * @throws {InvalidArgument} if the env is not valid
+ * @returns {string} - the corresponding OPeNDAP address
+ */
+function generateAddress() {
+  const env = process.env.CMR_ENVIRONMENT ? process.env.CMR_ENVIRONMENT.toLowerCase() : 'prod';
+
+  let envSubstition = env;
+  if (['prod', 'ops', 'uat', 'sit'].includes(env)) {
+    envSubstition = ((env === 'prod' || env === 'ops') ? '' : `${env}.`);
+  } else {
+    // Throw an exception if it is not a valid environment
+    throw new InvalidArgument(`Environment ${env} is not a valid environment.`);
+  }
+  return (`https://opendap.${envSubstition}earthdata.nasa.gov`);
+}
+
+/**
+ * getGranuleUr
+ *
+ * @param {string} metadata - the dom
+ * @param {boolean} isUmmG - UMM-G or ECHO10
+ * @returns {string} - the native id
+ */
+function getGranuleUr(metadata, isUmmG) {
+  let nativeId = null;
+  if (isUmmG === true) {
+    nativeId = metadata.GranuleUR;
+  } else {
+    const nativeIdNode = metadata.get('/Granule/GranuleUR');
+    nativeId = nativeIdNode.text();
+  }
+  return nativeId;
+}
+
+/**
+ * getEntryTitle
+ *
+ * @param {Object} config - comnfiguration
+ * @param {string} metadata - the granule metadata
+ * @param {boolean} isUmmG - whether this is UMM-G or ECHO10 metadata
+ * @returns {string} the entry title of the collection this granule belongs to
+ */
+async function getEntryTitle(config, metadata, isUmmG) {
+  let shortName = null;
+  let version = null;
+  if (isUmmG === true) {
+    shortName = metadata.CollectionReference.ShortName;
+    version = metadata.CollectionReference.Version;
+  } else {
+    shortName = metadata.get('/Granule/Collection/ShortName').text();
+    version = metadata.get('/Granule/Collection/VersionId').text();
+  }
+  // Query CMR for collection and retrieve entry title
+  const cmrInstance = new CMR({
+    provider: config.cmr.provider,
+    username: config.cmr.username,
+    password: config.cmr.passwordSecretName
+  });
+
+  const searchParams = {
+    short_name: shortName,
+    version: version
+  };
+
+  const result = await cmrInstance.searchCollections(searchParams);
+  // Verify that we have a valid result. If we don't then something is badly wrong
+  // and we should halt.
+  // Either the code is faulty or the provider is trying to ingest granules
+  // into a collection that doesn't exist
+  if (result.length === 0 || _.isUndefined(result[0].dataset_id)) {
```

Comment:
I think we could use `RecordDoesNotExist` error from the `errors` package for this.

Reply:
Updated